### PR TITLE
Sparfa admin settings

### PR DESCRIPTION
**Pull request by Dantemss**
- Change the Biglearn algorithm selection in the admin settings page to a dropdown select
- Make all courses use the chosen setting unless manually overridden

### CODE REVIEW

File: db/migrate/20170208182122_add_uuid_and_group_uuid_to_content_exercises.rb [hunk 1]
```diff
@@ -2,9 +2,10 @@ class AddUuidAndGroupUuidToContentExercises < ActiveRecord::Migration
   def change
     enable_extension 'pgcrypto'
 
-    add_column :content_exercises, :uuid, :uuid, null: false, index: true
+    add_column :content_exercises, :uuid, :uuid, null: false
     add_column :content_exercises, :group_uuid, :uuid, null: false
 
+    add_index :content_exercises, :uuid
     add_index :content_exercises, [:group_uuid, :version]
```
Comment (nathanstitt) [inline db/migrate/20170208182122_add_uuid_and_group_uuid_to_content_exercises.rb]: interesting.  what's the difference between `index: true` and `add_index` ?

File: spec/factories/course_profile/courses.rb [hunk 2]
```diff
@@ -18,12 +18,7 @@
 
     uuid                  { SecureRandom.uuid }
 
-    sequence_number                              { rand(1000) + 1 }
-    biglearn_student_clues_algorithm_name        { Faker::Hacker.abbreviation }
-    biglearn_teacher_clues_algorithm_name        { Faker::Hacker.abbreviation }
-    biglearn_assignment_spes_algorithm_name      { Faker::Hacker.abbreviation }
-    biglearn_assignment_pes_algorithm_name       { Faker::Hacker.abbreviation }
-    biglearn_practice_worst_areas_algorithm_name { Faker::Hacker.abbreviation }
+    sequence_number       { rand(1000) + 1 }
```
Comment (nathanstitt) [inline spec/factories/course_profile/courses.rb]: FYI, I recently learned `rand` accepts a range argument.  you can do: `rand(1..100)`

File: db/migrate/20170208182122_add_uuid_and_group_uuid_to_content_exercises.rb (hunk 1 above)
Comment (Dantemss) [reply db/migrate/20170208182122_add_uuid_and_group_uuid_to_content_exercises.rb]: `index: true` actually does not work with `add_column` (weird right?)

File: spec/factories/course_profile/courses.rb (hunk 2 above)
Comment (Dantemss) [reply spec/factories/course_profile/courses.rb]: Oh nice :) but I think I'll leave this one for now

**Pull request merged**